File: Latihan Kelas/KendaraanInheritance/Mobil.cpp
#include "Mobil.hpp"

// CONSTRUCTOR KELAS CHILD

// Default constructor untuk subclass Mobil
Mobil::Mobil() : Kendaraan(){
    this->supir = "XXX";
}

// User defined constructor untuk subclass Mobil
Mobil::Mobil(int nomor, int tahun, string merk, string supir) : Kendaraan(nomor, tahun, merk){
    this->supir = supir;
}

void Mobil::printInfo(){
    Kendaraan::printInfo();
    cout << "Kategori: Mobil" << endl;
    cout << "Supir: " << this->supir << endl;
}

int Mobil::biayaSewa(int lamaSewa){
    return lamaSewa * 50000;
}

File: Latihan Kelas/KendaraanInheritance - Jawwa/Mobil.java
public class Mobil extends Kendaraan {
    private String supir;

    public Mobil() {
        super();
        this.supir = "XXX";
    }

    public Mobil(int nomor, int tahun, String merk, String supir){
        super(nomor, tahun, merk);
        this.supir = supir;
    }

    public Mobil(Mobil m){
        super(m);
        this.supir = m.supir;
    }

    public void printInfo(){
        super.printInfo();
        System.out.println("Kategori: Mobil");
        System.out.println("Supir: " + supir);
    }

    public int biayaSewa(int lamaSewa){
        return lamaSewa * 50000;
    }
}
